ASIM/dev/ASimTester/filteringTest/filteringTest.py

```diff
@@ -0,0 +1,227 @@
+    return ','.join(paramsList)
+    
+
+def create_query_without_call(parser_file):
+    params_str = create_parameters_string(parser_file) 
+    query_from_yaml = parser_file['ParserQuery']
+    return f"let query= ({params_str}) {{ {query_from_yaml} }};\n"
+
+
+def create_call_without_parameter(column_name):
+    return f"query() | summarize count() by {column_name}\n" 
+
+
+def create_call_with_parameter(parameter, value, column_name):
+    return f"query({parameter}={value}) | summarize count() by {column_name}\n"
+
+
+class FilteringTest(unittest.TestCase):
+    # "Main" function which runs initiates the validations 
+    def tests_main_func(self):
+        if not os.path.exists(parser_file_path):
+            self.fail(f"File path does not exist: {parser_file_path}")
+        if not parser_file_path.endswith('.yaml'): 
+            self.fail(f"Not a yaml file: {parser_file_path}")
+        try:
+            parser_file = get_parser(parser_file_path)
+        except:
+            self.fail(f"Cannot open file: {parser_file_path}")
+        self.check_required_fields(parser_file)
+        no_call_query = create_query_without_call(parser_file)
+        self.check_data_in_workspace(no_call_query)
+        columns_in_answer = self.get_columns_of_parser_answer(no_call_query)
+        if parser_file['Normalization']['Schema'] not in all_schemas_parameters:
+            self.fail(f"Schema: {parser_file['Normalization']['Schema']} - is not supported")
+        param_to_column_mapping = all_schemas_parameters[parser_file['Normalization']['Schema']]
+        for param in parser_file['ParserParams']:
+            with self.subTest():
+                if param['Name'] not in param_to_column_mapping:
+                    self.fail(f"parameter: {param['Name']} - is not a valid parameter")
+                column_name_in_table = param_to_column_mapping[param['Name']]
+                self.handle_param(param, no_call_query, columns_in_answer, column_name_in_table)            
+                      
+    # Sending a parameters to the suitable test according to the parameter type    
+    def handle_param(self, param, no_call_query, columns_in_answer, column_name_in_table):
+        param_name = param['Name']
+        param_type = param['Type']
+        if (param_name == "pack"):
+            return 
+        if (param_name == "disabled"):
+            self.disabled_test(param, no_call_query)
+        elif  column_name_in_table not in columns_in_answer:
+            return
+        elif (param_type == "datetime"):
+            pass
+        elif (param_type == "dynamic"):
+            pass
+        else:
+            self.scalar_test(param, no_call_query, column_name_in_table)
+
+    
+    # Checking if the provided workspace has some data
+    def check_data_in_workspace(self, no_call_query):
+        check_data_response = self.send_query(no_call_query + "query() | take 5")
+        if len(check_data_response.tables[0].rows) == 0:
```

Comment:
>tables [](http://example.com/codeflow?start=35&length=6)

Will it return only one table? If check_data_response is null? Or tables is null?

Reply:
I couldn't find an example of a case when more than one table is returned. Also, when debugging I checked it and I always got only One table.
According to documentation, a query result won't be null but one of two types ("query_workspace" is used in "send_query":
![image](https://github.com/Azure/Azure-Sentinel/assets/126081432/0ca587a4-fdd4-4423-b366-31992004be1a)

Cases of failure or error during the query  are handled in "send_query".

Reply:
And what about the case where tables is null? Or no tables returned?

Reply:
Added an assertion for the cases you suggested in "send_query" function.